libs/partners/huggingface/langchain_huggingface/llms/huggingface_endpoint.py

```diff
@@ -172,15 +166,12 @@ def validate_environment(cls, values: Dict) -> Dict:
                 "Could not import huggingface_hub python package. "
                 "Please install it with `pip install huggingface_hub`."
             )
+        import os
 
-        values["huggingfacehub_api_token"] = get_from_dict_or_env(
-            values, "huggingfacehub_api_token", "HUGGINGFACEHUB_API_TOKEN", None
-        )
-
+        existing_key = next((key for key in ["HF_TOKEN", "HUGGINGFACEHUB_API_TOKEN"] if key in os.environ), "HUGGINGFACEHUB_API_TOKEN")
```

Comment:
@efriis this line will cause linting error, if needed I can do it in multiple lines